packages/core/src/typeschema/validation.test.ts

```diff
@@ -1535,6 +1535,115 @@ describe('FHIR resource validation', () => {
+      name: 'ClinicalDocument',
+      status: 'active',
+      kind: 'logical',
+      abstract: false,
+      type: 'http://hl7.org/cda/stds/core/StructureDefinition/CS',
+      snapshot: {
+        element: [
+          {
+            path: 'CS',
+          },
+          {
+            path: 'CS.code',
+            min: 0,
+            max: '1',
+            base: {
+              path: 'CD.code',
+              min: 0,
+              max: '1',
+            },
+            type: [{ code: 'code' }],
+          },
+        ],
+      },
+    };
+    loadDataType(csSd);
+
+    const clinicalDocumentSd: StructureDefinition = {
+      resourceType: 'StructureDefinition',
+      url: 'http://hl7.org/cda/stds/core/StructureDefinition/ClinicalDocument',
+      name: 'ClinicalDocument',
+      status: 'active',
+      kind: 'logical',
+      abstract: false,
+      type: 'http://hl7.org/cda/stds/core/StructureDefinition/ClinicalDocument',
+      snapshot: {
+        element: [
+          {
+            path: 'ClinicalDocument',
+          },
+          {
+            path: 'ClinicalDocument.realmCode',
+            min: 0,
+            max: '*',
+            base: {
+              path: 'ClinicalDocument.realmCode',
+              min: 0,
+              max: '*',
+            },
+            type: [
+              {
+                code: 'http://hl7.org/cda/stds/core/StructureDefinition/CS',
+              },
+            ],
+          },
+        ],
+      },
+    };
+    loadDataType(clinicalDocumentSd);
+
+    const value = { realmCode: [{ code: 'foo' }] };
+    const typedValue = { type: clinicalDocumentSd.type, value };
+    expect(() => validateTypedValue(typedValue)).not.toThrow();
+  });
+
+  test('Choice-of-type without [x]', () => {
+    // SubstanceAdministration.effectiveTime
+    // Source: https://github.com/HL7/CDA-core-sd/blob/master/input/resources/SubstanceAdministration.xml
+    // Note that there are multiple types for this property, but it does not have a [x] suffix
+    const sd: StructureDefinition = {
+      resourceType: 'StructureDefinition',
+      url: 'http://hl7.org/cda/stds/core/StructureDefinition/SubstanceAdministration',
+      name: 'SubstanceAdministration',
+      status: 'active',
+      kind: 'logical',
+      abstract: false,
+      type: 'http://hl7.org/cda/stds/core/StructureDefinition/SubstanceAdministration',
+      snapshot: {
+        element: [
+          {
+            path: 'SubstanceAdministration',
+          },
+          {
+            path: 'SubstanceAdministration.effectiveTime',
+            min: 0,
+            max: '*',
+            base: {
+              path: 'SubstanceAdministration.effectiveTime',
+              min: 0,
+              max: '*',
+            },
+            type: [{ code: 'dateTime' }, { code: 'time' }],
+          },
+        ],
+      },
+    };
+    loadDataType(sd);
+
+    const value = { effectiveTime: ['2022-02-02T12:00:00Z'] };
```

Comment:
Would it be possible to also add a case with a valid `time`, and an invalid type, just to cover the different permutations?